### DISCUSSION TITLE
Handover failure

### DISCUSSION THREAD

**Discussion by improperaffo** [srsran/srsRAN_Project#759] [General Help]
Dear all,
I am testing the new inter-du handover feature. I set up a CU and two DUs. My phones and modems can successfully connect to either one of the DUs, but if I try to force them to change DU with the `ho <serving pci> <rnti> <target pci>` command, the UE simply reconnects back to the same DU with a new RNTI. I briefly see a new RNTI in the other DU, but no traffic is ever exchanged. 
I looked through the RRC logs of the CU and everytime I see:
```
2024-08-05T14:18:40.450963 [RRC     ] [D] ue=2 c-rnti=0x4602: Containerized rrcReestablishmentRequest: [
  {
    "UL-CCCH-Message": {
      "message": {
        "c1": {
          "rrcReestablishmentRequest": {
            "rrcReestablishmentRequest": {
              "ue-Identity": {
                "c-RNTI": 17921,
                "physCellId": 2,
                "shortMAC-I": "0100011011110011"
              },
              "reestablishmentCause": "handoverFailure",
              "spare": "0"
            }
          }
        }
      }
    }
  }
]
```

What could be the cause?

Looking through the DU logs, it seems that handover routine is successful, but simultaneously I get a new UE trying to register with the old DU.

I tested with a Samsung phone, a OnePlus phone and a Quectel discrete modem.

[Here are the CU and DU logs](https://github.com/user-attachments/files/16498958/logs.zip)

**Comment by fllay** [comment]
I did the experiment for 2nd time and it seems to be ok. First of all you have to a bit careful about tx gain. If it is too small it will never connect. (I am using TX Gain about 50dB and RX Gain about 40dB and two cells are about 10 meters apart with a wall in between) and if it is too high (70dB 80dB), it will be on the first DU (I guess RX level is never low enough to switch DU).You can simply test by put the phone to airplane mode and turn it off when you are close to each cell to make sure that each cell is good.  Then you can walk with the phone between cells and it will be a while (30 secs or so) to make a switch. I am using Cellular-z and I can see NR-CI is switched between 1 and 2. However, it does not show neighboring cell info but I guess that now we are using the same gNB with two DUs (we are doing intra GNB handover anyway) so it does not show   neighboring cell.  I am using Samsung A33 phone.

**Reply by improperaffo** [reply]
Hi @fllay, thank you for your answer. The problem I am describing happens when I try to _force_ an handover, without relying on the metrics induced handover.

**Reply by improperaffo** [reply]
Hi @fllay , when you successfully triggered the measured-based handover, which SDRs were you using? And were they synchronized with an external clock?

**Reply by fllay** [reply]
I am using two B210s without any external clock and I can see from the gNB output when it can  successfully handover.

**Reply by improperaffo** [reply]
Thank you. I am stuck with this handover problem, so I was told the synchronization might be the problem.
[See this](https://github.com/srsran/srsRAN_Project/discussions/803#discussioncomment-10590484)

**Reply by pgawlowicz** [reply]
@improperaffo @fllay any update on this issue?
